src/ioctls/device.rs

```diff
@@ -80,6 +80,21 @@ impl DeviceFd {
         }
         Ok(())
     }
+
+    /// Gets a specified piece of device configuration and/or state.
+    ///
+    /// See the documentation for `KVM_GET_DEVICE_ATTR`.
+    /// # Arguments
+    ///
+    /// * `device_attr` - The device attribute to be get.
+    ///
+    pub fn get_device_attr(&self, device_attr: &mut kvm_device_attr) -> Result<()> {
```

Comment:
Hi @andreeaflorescu. It makes sense but I doubt a little bit if you can accept the following idea:

As far as I know, this ioctl works in this way: users put the `group` and `attr` into this `kvm_device_attr` structure as "requests", send it to the `KVM_GET_DEVICE_ATTR` ioctl, and the ioctl will return the information in a pointer which points to the "user-requested" data in the `addr` field.

In this way, the `kvm_device_attr` structure is both part of the input and part of the output. Maybe we can only return the `addr` field? I personally don't have specific preference.

Reply:
You are right @MrXinWang. The user needs to provide `group`, `attr` and a valid address (i.e `addr` field) otherwise the ioctl will fail.
As a side note, in many places (qemu included) KVM_GET_ATTR AND KVM_SET_ATTR are called through the same function.

Reply:
> As a side note, in many places (qemu included) KVM_GET_ATTR AND KVM_SET_ATTR are called through the same function.

Exactly, they used a bool parameter to determine whether it is read or write.

Reply:
Oh, I see.

The point would be to make this function (and every other one for that matter) easy to use, and hard to misuse. When you have parameters that work both as input and as output, that's hard to achieve.

Since it seems like some of the fields are used for input and some are for output, can you make the documentation more explicit? Also, a code example should be very useful. Most of the other functions have one. For the code example you don't need to do asserts (or use SET_DEVICE_ATTR). Just showcasing how you use the parameter as an input and as output should suffice.

Reply:
Hi @andreeaflorescu. I added the `#Returns` and the `#Examples` section in the doc and updated some details about the input and output arguments. Would you mind taking a look again?